runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -350,10 +350,6 @@ TR_VectorAPIExpansion::visitNodeToBuildVectorAliases(TR::Node *node)
          invalidateSymRef(child->getSymbolReference());
          }
       }
-   else if (node->getOpCodeValue() == TR::checkcast)
-      {
-      // ignore for now and check the children
-      }
```

Comment:
Was there a problem with `checkcast` here as well?

Reply:
yes, if the code during transformation is dead it's better not to ignore both checkcast and NULLCHK during the analysis. I will add a comment to this PR and will open a separate PR for those.

Reply:
done